e2e/tests/test_smoke.py

```diff
@@ -0,0 +1,146 @@
+
+        for filename in ["configmap.yaml", "postgres-operator.yaml"]:
+            utils.create_from_yaml(k8s_api.k8s_client, "manifests/" + filename)
+
+        # submit the most recent operator image built locally
+        # HACK assumes "images list" returns the most recent image at index 0
+        docker_client = docker.from_env()
+        image = docker_client.images.list(name="registry.opensource.zalan.do/acid/postgres-operator")[0].tags[0]
+        body = {
+            "spec": {
+                "template": {
+                    "spec": {
+                        "containers": [
+                            {
+                              "name": "postgres-operator",
+                              "image": image
+                            }
+                        ]
+                    }
+                 }
+            }
+        }
+        k8s_api.apps_v1.patch_namespaced_deployment("postgres-operator", "default", body)
+
+        # TODO check if CRD is registered instead
+        Utils.wait_for_pod_start(k8s_api, 'name=postgres-operator', cls.RETRY_TIMEOUT_SEC)
+
+        # HACK around the lack of Python client for the acid.zalan.do resource
+        subprocess.run(["kubectl", "create", "-f", "manifests/minimal-postgres-manifest.yaml"])
+
+        Utils.wait_for_pod_start(k8s_api, 'spilo-role=master', cls.RETRY_TIMEOUT_SEC)
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_master_is_unique(self):
+        """
+           Check that there is a single pod in the k8s cluster with the label "spilo-role=master".
+        """
+        k8s = K8sApi()
+        labels = 'spilo-role=master,version=acid-minimal-cluster'
+
+        num_of_master_pods = Utils.count_pods_with_label(k8s, labels)
+        self.assertEqual(num_of_master_pods, 1, "Expected 1 master pod, found {}".format(num_of_master_pods))
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_scaling(self):
+        """
+           Scale up from 2 to 3 pods and back to 2 by updating the Postgres manifest at runtime.
+        """
+        k8s = K8sApi()
+        labels = "version=acid-minimal-cluster"
+
+        Utils.wait_for_pg_to_scale(k8s, 3, self.RETRY_TIMEOUT_SEC)
+        self.assertEqual(3, Utils.count_pods_with_label(k8s, labels))
+
+        # TODO `kind` pods may stuck in the Terminating state for a few minutes; reproduce/file a bug report
+        Utils.wait_for_pg_to_scale(k8s, 2, self.RETRY_TIMEOUT_SEC)
+        self.assertEqual(2, Utils.count_pods_with_label(k8s, labels))
+
+
+class K8sApi:
+
+    def __init__(self):
+
+        # https://github.com/kubernetes-client/python/issues/309
+        warnings.simplefilter("ignore", ResourceWarning)
+
+        self.config = config.load_kube_config()
+        self.k8s_client = client.ApiClient()
+        self.core_v1 = client.CoreV1Api()
+        self.crd = client.CustomObjectsApi()
+        self.apps_v1 = client.AppsV1Api()
```

Comment:
We will need more APIs, don't we? Well, can be added when needed...

Reply:
that's exactly the idea behind this class: collect APIs in one place and add as needed